collection/collection.go

```diff
@@ -97,11 +96,20 @@ func (sm *SetMaintainer) handleEvent(ctx context.Context, e event.Event) error {
 		// keep in mind commit changes can mean added OR removed versions
 		if vi, ok := e.Payload.(dsref.VersionInfo); ok {
 			sm.UpdateEverywhere(ctx, vi.InitID, func(m *dsref.VersionInfo) {
-				// preserve workflow id
+				// preserve fields that are not tracked in `ETDatasetCommitChanges`
 				vi.WorkflowID = m.WorkflowID
-				// preserve run information
-				vi.RunID = m.RunID
-				vi.RunStatus = m.RunStatus
+				vi.DownloadCount = m.DownloadCount
+				vi.RunCount = m.RunCount
+				vi.FollowerCount = m.FollowerCount
+				vi.OpenIssueCount = m.OpenIssueCount
+
+				// preserve "last run" information
+				if vi.RunID == "" {
+					vi.RunID = m.RunID
+					vi.RunStatus = m.RunStatus
+					vi.RunDuration = m.RunDuration
+				}
+
```

Comment:
Great catch, and _very_ interesting. This is starting to hint at a separate struct, but nothing we should act on yet IMHO.